app/components/Form/TextInput.tsx

```diff
@@ -1,40 +1,70 @@
 import cx from 'classnames';
+import { useRef } from 'react';
+import Icon from 'app/components/Icon';
+import Flex from 'app/components/Layout/Flex';
 import { createField } from './Field';
 import styles from './TextInput.css';
-import type { RefObject, InputHTMLAttributes } from 'react';
+import type { RefObject, ReactNode, InputHTMLAttributes } from 'react';
 
 type Props = {
   type?: string;
+  prefix?: ReactNode;
   suffix?: string;
   className?: string;
-  inputRef?: RefObject<HTMLInputElement>;
+  inputRef?: HTMLInputElement;
   disabled?: boolean;
   readOnly?: boolean;
+  placeholder?: string;
+  removeBorder?: boolean;
 } & InputHTMLAttributes<HTMLInputElement>;
 
-function TextInput({
+const TextInput = ({
   type = 'text',
   className,
   disabled,
   inputRef,
+  prefix,
   suffix,
   readOnly,
+  placeholder,
+  removeBorder = false,
   ...props
-}: Props) {
+}: Props) => {
+  /* New ref is made because text inputs that are not Fields are not given a ref */
+  const newInputRef = useRef<HTMLInputElement>(inputRef);
+
   return (
-    <span className={cx(suffix && styles.inputGroup)}>
+    <Flex
+      alignItems="center"
+      gap={10}
+      style={{ border: removeBorder && 'none' }}
```

Comment:
Isn't inline css a bad idea ??

Reply:
Not in this case, as it's dynamic and directly related to the prop.